lib/debug/deprecate.js

```diff
@@ -34,6 +35,16 @@ const assert = require('./assert');
  * - `url` A URL that refers to additional information about the deprecation
  */
 function deprecate(description, condition, options) {
+  if (options && options.for) {
```

Comment:
This should be moved under:

```js
if (condition) {
  return;
}
```

Because we want to assert all options first, that way you don't need the if here, and we want to take the `condition` value into account I think.

Reply:
I agree that the check should move below the assert calls in the lines below.

Reply:
This should also move below the `if (condition)` as Bert says -- the condition is whether or not the deprecation is triggered, so we wouldn't want to throw on ones that are not triggered.

Reply:
yes, made the changes